storehaus-kafka/src/main/scala/com/twitter/storehaus/kafka/JavaFutureToTwitterFutureConverter.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2014 Twitter inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.storehaus.kafka
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{Future => JFuture}
+
+import com.twitter.util.{Future, Try, Promise}
+
+import scala.annotation.tailrec
+
+/**
+  * Utility class for converting Java futures to Twitter's 
+  * @param waitTimeMs Time spent sleeping by the thread converting java futures to
+  *                   twitter futures when there are no futures to convert in ms
+  */
+private[kafka] class JavaFutureToTwitterFutureConverter(waitTimeMs: Long = 1000L) {
+  
+  def apply[T](javaFuture: JFuture[T]): Future[T] = {
+    val promise = new Promise[T]()
+    poll(Link(javaFuture, promise))
+    promise
+  }
+  
+  private val pollRun = new Runnable {
+    override def run(): Unit =
+      try {
+        while (!Thread.currentThread().isInterrupted)
+          loop(list.getAndSet(Nil))
+      } catch {
+        case e: InterruptedException =>
+      }
+    
+    @tailrec
+    def loop(links: List[Link[_]]): Unit = {
+      val notDone = links.filterNot(_.maybeUpdate)
+      if (links.isEmpty || notDone.nonEmpty) Thread.sleep(waitTimeMs)
+      loop(list.getAndSet(notDone))
+    }
+  }
+  private val list = new AtomicReference[List[Link[_]]](Nil)
+  private val thread = new Thread(pollRun)
+  
+  def start(): Unit = {
+    thread.setDaemon(true)
+    thread.start()
+  }
+  
+  def stop(): Unit = {
+    list.get().foreach { l =>
+      l.promise.setException(new Exception("Promise not completed"))
+      l.future.cancel(true)
+    }
+    thread.interrupt()
+  }
+
+  private def poll[T](link: Link[T]): Unit = {
```

Comment:
there is a race here if some one calls apply after stop. That could be okay if we advertise that they must be synchronized, or we could handle by having a bigger state:
```
sealed abstract class State
case object Closed extends State
case class Open(links: List[Link[_]]) extends State
val EmptyOpen: State = Open(Nil)

@annotation.tailrec
def swapOpen(old: List[Link[_]]): (List[Link[_]], Boolean) = list.get match {
  case Closed => (old, false)
  case s@Open(items) = if (list.compareAndSet(s, Open(old))) (items, true) else takeOpen
}

def poll[T](l: Link[T]): Unit = list.get match {
  case Closed => // cancel and complete
  case s@Open(tail) => if (list.compareAndSet(s, Open(l :: tail)) () else poll(l)
}

def stop(): Unit = {
  list.getAndSet(Closed) match {
    case Closed => // already closed
    case Open(ls) => // close all the futures.
  }
}
```